File: runtime/python/src/mica/health.py
from __future__ import annotations

from dataclasses import dataclass

from mica.errors import RpcCode


class State:
    UNSPECIFIED = "UNSPECIFIED"
    STARTING = "STARTING"
    RUNNING = "RUNNING"
    STOPPING = "STOPPING"
    STOPPED = "STOPPED"
    FAILED = "FAILED"


@dataclass
class Health:
    state: str
    message: str
    uptime_ms: int


def rpc_code_name(code: RpcCode) -> str:
    return RpcCode(code).name

File: runtime/go/mica/health.go
package mica

type State string

const (
	StateUnspecified State = "UNSPECIFIED"
	StateStarting    State = "STARTING"
	StateRunning     State = "RUNNING"
	StateStopping    State = "STOPPING"
	StateStopped     State = "STOPPED"
	StateFailed      State = "FAILED"
)

type Health struct {
	State    State
	Message  string
	UptimeMS uint64
}
